### PR TITLE
apache healthcheck, check localhost apacheport with every apache port

### PR DESCRIPTION
**Pull request by Zoey2936**

### CODE REVIEW

**Comment by szaimen** [comment]
Thanks Zoey! Is there a reason for this change?

**Comment by Zoey2936** [comment]
> Thanks Zoey! Is there a reason for this change?

No... But I thought that localhost can be checked every time

**Comment by Zoey2936** [comment]
Other question, AIO only supports offially port 443, what about also alwas checking port 443 from external?

**Comment by szaimen** [comment]
> Other question, AIO only supports offially port 443, what about also alwas checking port 443 from external?

That would basically mean that the healthcheck would break behind cloudflare, IIRC?

**Comment by szaimen** [comment]
> > Thanks Zoey! Is there a reason for this change?
> 
> No... But I thought that localhost can be checked every time

Does this have any benefit? (I mean if the domain and 443 works, the container should be healthy, no?

**Comment by Zoey2936** [comment]
> 

why should it break? cloudflare has port 443 open

**Comment by Zoey2936** [comment]
> > > Thanks Zoey! Is there a reason for this change?
> > 
> > 
> > No... But I thought that localhost can be checked every time
> 
> Does this have any benefit? (I mean if the domain and 443 works, the container should be healthy, no?

if it runs behind cloudflare the port is open, but the apache container could be stopped

**Comment by szaimen** [comment]
> > 
> 
> why should it break? cloudflare has port 443 open

In the past were connection attempts blocked by cloudflare... but you are probably right because this only checks if a tcp connection is possible which is likely not blocked by cloudflare.

**Comment by szaimen** [comment]
However what about scenarios where like when using dns challenge where the port will not be opened?

**Review by szaimen — APPROVED** [review]
Fine by me in the current form

**Comment by Zoey2936** [comment]
> However what about scenarios where like when using dns challenge where the port will not be opened?

you mean in local instances? even they need to have port 443 open and the nextcloud domain should point to the domain

**Comment by szaimen** [comment]
> you mean in local instances? even they need to have port 443 open and the nextcloud domain should point to the domain

True. Okay then lets always check it

File: Containers/apache/healthcheck.sh [hunk 1]
```diff
@@ -2,8 +2,5 @@
 
 nc -z "$NEXTCLOUD_HOST" 9000 || exit 0
 nc -z localhost 8000 || exit 1
-if [ "$APACHE_PORT" != '443' ]; then
-    nc -z localhost "$APACHE_PORT" || exit 1
-else
-    nc -z "$NC_DOMAIN" "$APACHE_PORT" || exit 1
-fi
+nc -z localhost "$APACHE_PORT" || exit 1
+nc -z "$NC_DOMAIN" "$APACHE_PORT" || exit 1
```
Comment (szaimen) [inline Containers/apache/healthcheck.sh:6]:
```suggestion
nc -z "$NC_DOMAIN" 443 || exit 1
```

**Review by szaimen — CHANGES_REQUESTED** [review]

**Review by szaimen — APPROVED** [review]

**Comment by szaimen** [comment]
This is now released with v4.9.0 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel

**Pull request merged**